File: 1642-WaterBottles/1642-WaterBottles.cpp
// Last updated: 2/11/2026, 11:08:18 AM
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int fullBottles = numBottles;
        int emptyBottles = 0;
        int total = 0;

        while (fullBottles > 0){
            total += fullBottles;

            emptyBottles += fullBottles;

            fullBottles = emptyBottles/numExchange;
            emptyBottles = emptyBottles%numExchange;
        }

        return total;
    }
};

File: 1642-WaterBottles/1642-WaterBottles.py
# Last updated: 2/11/2026, 11:08:21 AM
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        fullBottles = numBottles
        emptyBottles = 0

        total = 0
        
        while (fullBottles > 0):
            #drink the full bottles
            total+= fullBottles
            
            #now the dranked fullbottles are empty bottles

            emptyBottles += fullBottles

            #exchange emptybottles for fullbottles
            fullBottles = emptyBottles//numExchange
            #they may be some empty bottle remain
            emptyBottles = emptyBottles%numExchange


        return total

        